sdstore/sdstore.go

```diff
@@ -132,103 +122,93 @@ func (s *sdStore) GenerateAndCheckMd5Json(url *url.URL, path string) (string, er
 	}
 
 	jsonString, err := json.Marshal(newMd5)
-
 	if err != nil {
 		return "", err
 	}
 
-	var md5Path string
-	md5Path = fmt.Sprintf("%s_md5.json", filepath.Base(path))
+	md5Path := fmt.Sprintf("%s_md5.json", filepath.Base(path))
 	jsonFile, err := os.Create(md5Path)
-
 	if err != nil {
 		return "", err
 	}
 	defer jsonFile.Close()
 
 	jsonFile.Write(jsonString)
-	jsonFile.Close()
 
 	return md5Path, nil
 }
 
 // Uploads sends a file to a path within the SD Store. The path is relative to
 // the build/event path within the SD Store, e.g. http://store.screwdriver.cd/builds/abc/<storePath>
 func (s *sdStore) Upload(u *url.URL, filePath string, toCompress bool) error {
-	var err error
-
-	for i := 0; i < s.maxRetries; i++ {
-		time.Sleep(time.Duration(float64(i*i)*s.retryScaler) * time.Second)
-
-		if toCompress {
-			var fileName string
-			fileName = filepath.Base(filePath)
-			encodedURL, _ := url.Parse(fmt.Sprintf("%s%s", u.String(), "_md5.json"))
-			md5Json, err := s.GenerateAndCheckMd5Json(encodedURL, filePath)
-
-			if err != nil && err.Error() == "Contents unchanged" {
-				log.Printf("No change to %s, aborting upload", filePath)
-				return nil
-			}
-
-			if err != nil {
-				log.Printf("(Try %d of %d) error received from generating md5: %v", i+1, s.maxRetries, err)
-				continue
-			}
-
-			err = s.putFile(encodedURL, "application/json", md5Json)
-			if err != nil {
-				log.Printf("(Try %d of %d) error received from uploading md5 json: %v", i+1, s.maxRetries, err)
-				continue
-			}
-
-			err = os.Remove(md5Json)
-			if err != nil {
-				log.Printf("Unable to remove md5 file from path: %s, continuing", md5Json)
-			}
-
-			var zipPath string
-			zipPath, err = filepath.Abs(fmt.Sprintf("%s.zip", fileName))
-
-			if err != nil {
-				log.Printf("(Try %d of %d) Unable to determine filepath: %v", i+1, s.maxRetries, err)
-				continue
-			}
+	if !toCompress {
```

Comment:
`!toCompress` codes are smaller than `toCompress`. Thus I flip the order of `toCompress` and `!toCompress`.